### ISSUE TITLE
na_ontap_user fails on existing user with same password

### ISSUE THREAD

**Issue by chjones45** [ansible-collections/netapp#42]
##### SUMMARY
na_ontap_user module fails if the user already exists with the same password

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
na_ontap_user

##### ANSIBLE VERSION

```
ansible 2.9.7
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/usr/share/my_modules']
  ansible python module location = /usr/lib/python2.7/dist-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.17 (default, Apr 15 2020, 17:20:14) [GCC 7.5.0]
```

##### CONFIGURATION

```
DEFAULT_MODULE_PATH(/etc/ansible/ansible.cfg) = [u'/usr/share/my_modules']
DEFAULT_ROLES_PATH(/etc/ansible/ansible.cfg) = [u'/etc/ansible/roles']
DEPRECATION_WARNINGS(/etc/ansible/ansible.cfg) = False
```

##### OS / ENVIRONMENT
FAS8060 running ONTAP 9.7P3

##### STEPS TO REPRODUCE
Issue the na_ontap_user task when the user already exists with the correct password.

```
- name: Create ONTAP User for ActiveIQ Unified Manager
  netapp.ontap.na_ontap_user:
    <<: *clusterlogin
    state: "{{ item.state }}"
    vserver: "{{ cluster_name }}"
    name: "{{ item.ontap_user }}"
    applications: ontapi,ssh,http
    authentication_method: password
    set_password: "{{ item.ontap_user_password }}"
    role_name: admin
  with_items:
    "{{ activeiq_unified_manager }}"
  when:
    activeiq_unified_manager != None
  tags: aiqum_user
```

##### EXPECTED RESULTS
The module should handle the "Error while updating user password: {'message': 'New password must be different than the old password.', 'code': '7077925'}" response from ONTAP as being an acceptable result.

##### ACTUAL RESULTS
```
<fas-cluster1> EXEC /bin/sh -c 'echo ~root && sleep 0'
<fas-cluster1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /root/.ansible/tmp `"&& mkdir /root/.ansible/tmp/ansible-tmp-1591315679.77-4065-176919232991861 && echo ansible-tmp-1591315679.77-4065-176919232991861="` echo /root/.ansible/tmp/ansible-tmp-1591315679.77-4065-176919232991861 `" ) && sleep 0'
Using module file /root/.ansible/collections/ansible_collections/netapp/ontap/plugins/modules/na_ontap_user.py
<fas-cluster1> PUT /root/.ansible/tmp/ansible-local-40263floEB/tmpPZW3ii TO /root/.ansible/tmp/ansible-tmp-1591315679.77-4065-176919232991861/AnsiballZ_na_ontap_user.py
<fas-cluster1> EXEC /bin/sh -c 'chmod u+x /root/.ansible/tmp/ansible-tmp-1591315679.77-4065-176919232991861/ /root/.ansible/tmp/ansible-tmp-1591315679.77-4065-176919232991861/AnsiballZ_na_ontap_user.py && sleep 0'
<fas-cluster1> EXEC /bin/sh -c '/usr/bin/python3 /root/.ansible/tmp/ansible-tmp-1591315679.77-4065-176919232991861/AnsiballZ_na_ontap_user.py && sleep 0'
<fas-cluster1> EXEC /bin/sh -c 'rm -f -r /root/.ansible/tmp/ansible-tmp-1591315679.77-4065-176919232991861/ > /dev/null 2>&1 && sleep 0'
failed: [fas-cluster1] (item={u'applications': u'ssh,http,ontapi', u'state': u'present', u'role': u'admin', u'name': u'chris', u'password': u'netapp123', u'auth_method': u'password'}) => {
    "ansible_loop_var": "item",
    "changed": false,
    "invocation": {
        "module_args": {
            "applications": [
                "ssh",
                "http",
                "ontapi"
            ],
            "authentication_method": "password",
            "authentication_password": null,
            "authentication_protocol": null,
            "cert_filepath": null,
            "engine_id": null,
            "feature_flags": {},
            "hostname": "10.128.58.113",
            "http_port": null,
            "https": true,
            "key_filepath": null,
            "lock_user": null,
            "name": "scommonitor",
            "ontapi": null,
            "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "privacy_password": null,
            "privacy_protocol": null,
            "remote_switch_ipaddress": null,
            "role_name": "admin",
            "set_password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "state": "present",
            "use_rest": "Auto",
            "username": "admin",
            "validate_certs": false,
            "vserver": "fas-cluster1"
        }
    },
    "item": {
        "applications": "ssh,http,ontapi",
        "auth_method": "password",
        "name": "chris",
        "password": "netapp123",
        "role": "admin",
        "state": "present"
    },
    "msg": "Error while updating user password: {'message': 'New password must be different than the old password.', 'code': '7077925'}"
```

**Comment by lonico** [comment]
Yes, it's an issue we introduced when adding REST support.
A temporary workaround can be to use `use_rest: Never`.

**Comment by carchi8py** [comment]
@chjones45 we have DEVOPS-2782 open for this in our Jira instance to fix this for REST

**Comment by chjones45** [comment]
Confirmed this error is resolved in the 20.6.1 update. Much appreciated!
